### ISSUE TITLE
Newly created FIDB not available next iteration

### ISSUE THREAD

**Issue by qrkyckling** [NationalSecurityAgency/ghidra#4142]
I have create a small Python script to create an empty Fid file.

```python
# Open output file
        ....
def run(self):
        Outfile = java.io.File(self.args["OutfilePath"])
        print("Creating empty FIDB -> {}".format(Outfile.getName() ) )

        try:
            FFM._FFM.createNewFidDatabase( Outfile )
        except:
            print("FAILED to create {}. Does it already exist ?".format(Outfile.getName() ) )
 
        # Add FID file to project
        fidFile = self._FFM.addUserFidFile( Outfile )
        fidFile.setActive(True)
        fidDB = fidFile.getFidDB(True)
        fidDB.saveDatabase("Saving", monitor)
        fidDB.close()
       ....

```

I am running this in headless mode and if I pass it an argument of "newfid.fidb" I do indeed get a file of about 668 bytes called newfid.fidb.
I add it to the project above using addUserFidFile() and I verified it is added by using getUserAddedFiles().
It is listed among the added/attached fids.

Next step I try to call CreateMultipleLibraries.java using my newly created newfid.fidb but it returns
**"Invalid choice: newfid.fidb (HeadlessAnalyzer) java.lang.IllegalArgumentException: Invalid choice: newfid.fidb"**

Opening the project manually I can confirm that "newfid.fidb" is no longer among the currently attached fidbs (and not available in getUserAddedFiles() which I believe to be the problem.

Why after I ran addUserFidFile() on newfid.fidb it is not attached and available for use in CreateMultipleLibraries.java ?
Do I need to force save the project somehow ?

**Comment by qrkyckling** [comment]
Chaining my "GhidraCreateEmptyFidb.py" as a preScript and then running CreateMultipleLibraries.java as postScript I get action happening in /tmp/user-Ghidra/packed-db-cache/pdb..../

Running only GhidraCreateEmptyFidb.py gives no action.
Running only CreateMultipleLibraries.java also gives no action.

So chaining them together gives no error. And gives a lot of action in the cache dir.
A file of 49kb is left in there.
But still my newfid.fidb is left at 667 bytes.
Sometimes becomes 668 or 669 bytes.

So it seems whatever processes are supposed to work are working but the actual saving is not happening ?
Goes both for saving my newfid.fidb and saving the user added fidFiles to the properties.

**Comment by qrkyckling** [comment]
Got around to have another look at this.
While the FIDB is still not added to the project I figured out what was causing it to never get populated.

Reading CreateMultipleLibraries.java I found the comment at createLibraryNames() saying that I need to have a certain structure of the added folders.

```
	private void createLibraryNames() {
		// path should look like : compiler, project, version, options
		currentLibraryName = pathelement[1];
		currentLibraryVersion = pathelement[2];
		currentLibraryVariant = pathelement[0] + ':' + pathelement[3];
	}
```

Creating a structure like this I had my FIDB populated.
Unfortunately there were no errors or warnings on the console about this.

Closing this issue since I figured out what was wrong.
